Travis fails (Python1.4)

**Describe the bug**
It seems that -out of nowhere-, Travis CI is failing again.

**Screenshots and logs**
One of the Travis CI logs is [here](https://travis-ci.com/github/GQCG/GQCP/builds/179222616), and here is the relevant part of the log:

> -- Found PythonInterp: /opt/pyenv/shims/python3.7 (found version "1.4") 
> 377CMake Error at /home/travis/miniconda/share/cmake/pybind11/FindPythonLibsNew.cmake:96 (message):
> 378  Python config failure:
> 379
> 380  pyenv: python3.7: command not found
> 381
> 382  
> 383
> 384  The `python3.7' command exists in these Python versions:
> 385
> 386    3.7
> 387    3.7.6

--- a/.travis.yml
+++ b/.travis.yml
@@ -72,7 +72,7 @@ install:
 script:
   - cd ${TRAVIS_BUILD_DIR}
   - mkdir build && cd build
-  - cmake .. -DBUILD_TESTS=TRUE -DBUILD_BENCHMARKS=TRUE -DBUILD_PYTHON_BINDINGS=TRUE -DCMAKE_INSTALL_PREFIX=~/.local -DCMAKE_PREFIX_PATH=${HOME}/miniconda-DPYTHON_EXECUTABLE=${HOME}/miniconda/bin/python3.7-DPYTHON_LIBRARY=${HOME}/miniconda/lib/libpython3.7m.so -DPYTHON_INCLUDE_DIR=${HOME}/miniconda/include/python3.7m
+  - cmake .. -DBUILD_TESTS=TRUE -DBUILD_BENCHMARKS=TRUE -DBUILD_PYTHON_BINDINGS=TRUE -DCMAKE_INSTALL_PREFIX=~/.local -DCMAKE_PREFIX_PATH=${HOME}/miniconda -DPYTHON_EXECUTABLE=${HOME}/miniconda/bin/python3.7 -DPYTHON_LIBRARY=${HOME}/miniconda/lib/libpython3.7m.so -DPYTHON_INCLUDE_DIR=${HOME}/miniconda/include/python3.7m
   - make -j3 VERBOSE=1 && env CTEST_OUTPUT_ON_FAILURE=1 make test ARGS=-j3 && sudo make install
 
 # Run the benchmarks

--- a/.travis.yml
+++ b/.travis.yml
@@ -72,15 +72,7 @@ install:
 script:
   - cd ${TRAVIS_BUILD_DIR}
   - mkdir build && cd build
-  - cmake .. 
-  -DBUILD_TESTS=TRUE 
-  -DBUILD_BENCHMARKS=TRUE 
-  -DBUILD_PYTHON_BINDINGS=TRUE 
-  -DCMAKE_INSTALL_PREFIX=~/.local 
-  -DCMAKE_PREFIX_PATH=${HOME}/miniconda
-  -DPYTHON_EXECUTABLE=${HOME}/miniconda/bin/python3.7
-  -DPYTHON_LIBRARY=${HOME}/miniconda/lib/libpython3.7m.so 
-  -DPYTHON_INCLUDE_DIR=${HOME}/miniconda/include/python3.7m
+  - cmake .. -DBUILD_TESTS=TRUE -DBUILD_BENCHMARKS=TRUE -DBUILD_PYTHON_BINDINGS=TRUE -DCMAKE_INSTALL_PREFIX=~/.local -DCMAKE_PREFIX_PATH=${HOME}/miniconda-DPYTHON_EXECUTABLE=${HOME}/miniconda/bin/python3.7-DPYTHON_LIBRARY=${HOME}/miniconda/lib/libpython3.7m.so -DPYTHON_INCLUDE_DIR=${HOME}/miniconda/include/python3.7m
   - make -j3 VERBOSE=1 && env CTEST_OUTPUT_ON_FAILURE=1 make test ARGS=-j3 && sudo make install
 
 # Run the benchmarks

--- a/.travis.yml
+++ b/.travis.yml
@@ -72,7 +72,15 @@ install:
 script:
   - cd ${TRAVIS_BUILD_DIR}
   - mkdir build && cd build
-  - cmake .. -DBUILD_TESTS=TRUE -DBUILD_BENCHMARKS=TRUE -DBUILD_PYTHON_BINDINGS=TRUE -DCMAKE_INSTALL_PREFIX=~/.local -DCMAKE_PREFIX_PATH=${HOME}/miniconda
+  - cmake .. 
+  -DBUILD_TESTS=TRUE 
+  -DBUILD_BENCHMARKS=TRUE 
+  -DBUILD_PYTHON_BINDINGS=TRUE 
+  -DCMAKE_INSTALL_PREFIX=~/.local 
+  -DCMAKE_PREFIX_PATH=${HOME}/miniconda
+  -DPYTHON_EXECUTABLE=${HOME}/miniconda/bin/python3.7
+  -DPYTHON_LIBRARY=${HOME}/miniconda/lib/libpython3.7m.so 
+  -DPYTHON_INCLUDE_DIR=${HOME}/miniconda/include/python3.7m
   - make -j3 VERBOSE=1 && env CTEST_OUTPUT_ON_FAILURE=1 make test ARGS=-j3 && sudo make install
 
 # Run the benchmarks

--- a/.travis.yml
+++ b/.travis.yml
@@ -5,6 +5,8 @@ sudo: true
 # Specify the project language and the compiler to test to
 language: cpp
 compiler: gcc
+python: 
+  - "3.7"
 
 addons:
   apt:

--- a/.travis.yml
+++ b/.travis.yml
@@ -42,7 +42,7 @@ before_install:
         if [[ "${TRAVIS_PYTHON_VERSION}" == "2.7" ]]; then
             wget https://repo.continuum.io/miniconda/Miniconda2-latest-Linux-x86_64.sh -O $HOME/download/miniconda.sh;
         else
-            wget https://repo.continuum.io/miniconda/Miniconda3-latest-Linux-x86_64.sh -O $HOME/download/miniconda.sh;
+            wget https://repo.continuum.io/miniconda/Miniconda3-py37_4.8.3-Linux-x86_64.sh -O $HOME/download/miniconda.sh;
         fi;
         bash $HOME/download/miniconda.sh -b -p ${HOME}/miniconda;
       fi

